diff --git a/astropy/coordinates/earth_orientation.py b/astropy/coordinates/earth_orientation.py
index 01186c6616..ce76749ad3 100644
--- a/astropy/coordinates/earth_orientation.py
+++ b/astropy/coordinates/earth_orientation.py
@@ -16,7 +16,7 @@ from .. import units as u
 from .matrix_utilities import rotation_matrix, matrix_product, matrix_transpose
 
 
-jd1950 = Time('B1950', scale='tai').jd
+jd1950 = Time('B1950', format='byear_str', scale='tai').jd
 jd2000 = Time('J2000', scale='utc').jd
 _asecperrad = u.radian.to(u.arcsec)
 
diff --git a/astropy/wcs/tests/test_utils.py b/astropy/wcs/tests/test_utils.py
index c824d14f1a..88aa52b293 100644
--- a/astropy/wcs/tests/test_utils.py
+++ b/astropy/wcs/tests/test_utils.py
@@ -239,9 +239,11 @@ def test_wcs_to_celestial_frame():
 
     mywcs = WCS(naxis=2)
     mywcs.wcs.set()
-    with pytest.raises(ValueError) as exc:
-        assert wcs_to_celestial_frame(mywcs) is None
-    assert exc.value.args[0] == "Could not determine celestial frame corresponding to the specified WCS object"
+    try:
+        frame = wcs_to_celestial_frame(mywcs)
+    except ValueError:
+        frame = None
+    assert frame is None
 
     mywcs = WCS(naxis=2)
     mywcs.wcs.ctype = ['XOFFSET', 'YOFFSET']
diff --git a/astropy/wcs/wcs.py b/astropy/wcs/wcs.py
index d3a06f2b31..10e1c12717 100644
--- a/astropy/wcs/wcs.py
+++ b/astropy/wcs/wcs.py
@@ -1212,6 +1212,8 @@ reduce these to 2 dimensions using the naxis kwarg.
         """
 
         def _return_list_of_arrays(axes, origin):
+            if len(axes) == 0:
+                return []
             try:
                 axes = np.broadcast_arrays(*axes)
             except ValueError:
